src/main/java/com/kruize/recommendations/application/ApplicationRecommendationsImpl.java

```diff
@@ -178,16 +178,23 @@ public double getCpuRequests(String applicationName) throws NoSuchApplicationExc
         }
     }
 
-    /* If all the instances of the application are idle, return idle */
+    /* If any instances of the application is running, return running */
     public String getStatus(String applicationName)
     {
+        boolean applicationIdle = false;
         for (MetricsImpl metric: applicationMap.get(applicationName))
         {
             if (metric.getStatus().equals("running"))
                 return "running";
+
+            if (metric.getStatus().equals("idle"))
+                applicationIdle = true;
         }
 
-        return "idle";
+        if (applicationIdle)
+            return "idle";
+
+        return applicationMap.get(applicationName).get(0).getStatus();
```

Comment:
Can you point me what are all the valid states for application status ?

Reply:
Idle, running, pending and terminated.

Reply:
I meant where is documented in the code, along with comments on what it means.